Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/recovery/core/RecoveryCertificateRepository.kt

```diff
@@ -114,6 +115,14 @@ class RecoveryCertificateRepository @Inject constructor(
             scope = appScope
         )
 
```

Comment:
_allcertficates_ is already doing this mapping, why do it here again 😸 ?

Reply:
when we register a certificate `internalData` flows report that the certificate is registered and returns its `containerId` and this container ID is used to navigate to the details. `allcertficates` is flow based on another flow and some others flows if many certificates are already in the App it takes time to emit the latest certificates while we are sure that the certificate is just registered, `allcertficates` does not have it yet. 

TL;DR
If we are rely on `internalData` flow to know if the certificate is registered, `internalData` should be the source to get the Certificate from

Reply:
all right